pumpkin/src/server/translation.rs

```diff
@@ -0,0 +1,208 @@
+#![expect(dead_code)]
+
+use std::{
+    collections::HashMap,
+    fs::File,
+    io::{BufRead, BufReader},
+};
+
+use pumpkin_config::TranslationConfig;
+use pumpkin_core::text::{style::Style, TextComponent, TextContent};
+use serde_json::Value;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum TranslationError {
+    #[error("File cannot be opened.")]
+    InvalidFile,
+    #[error("Failed to read file. Error: {0}")]
+    FileRead(std::io::Error),
+    #[error("Invalid JSON encountered. Use only objects in the translation file.")]
+    JsonParse,
+}
+
+pub enum Translations<'a> {
+    Untranslated(TextComponent<'a>),
+    Translated(HashMap<String, TextComponent<'a>>),
```

Comment:
Im also not sure why it has a HashMap, Both should just return a TextComponent

Reply:
HashMap is in case more than 1 string contains the message. Do I just check for exact matches instead?